src/output-json-dns.c

```diff
@@ -304,6 +304,21 @@ JsonBuilder *JsonDNSLogAnswer(void *txptr, uint64_t tx_id)
     }
 }
 
+bool AlertJsonDns(void *txptr, JsonBuilder *js)
```

Comment:
should this return false if both calls return NULL?

Reply:
Likely so, this is a behavioral change

Reply:
> Likely so, this is a behavioral change

Not a big one right?  It would allow the caller to reset to mark to prevent an empty `dns` object from being logger in alert.

Reply:
Does not look a big one 

#9005 is the part without behavior changes

Then I intend to do the behavioral changes about alerts : addition of protocols missing alert metadata (like krb5) + behavioral change for dns alert metadata